### ISSUE TITLE
BEVDet splash过程与LSS实现过程不匹配

### ISSUE THREAD

**Issue by northeastsquare** [HuangJunJie2017/BEVDet#379]
LSS:https://github.com/nv-tlabs/lift-splat-shoot/blob/master/src/models.py#L235
        # griddify (B x C x Z x X x Y)
        final = torch.zeros((B, C, self.nx[2], self.nx[0], self.nx[1]), device=x.device)
        final[geom_feats[:, 3], :, geom_feats[:, 2], geom_feats[:, 0], geom_feats[:, 1]] = x
final特征是BCZXY, 所以final = torch.cat(final.unbind(dim=2), 1)，是把高度Z维度合并到C特征维

而Bevdet: https://github.com/HuangJunJie2017/BEVDet/blob/dev3.0/mmdet3d/models/necks/view_transformer.py#L225
bev_feat_shape = (depth.shape[0], int(self.grid_size[2]),
                          int(self.grid_size[1]), int(self.grid_size[0]),
                          feat.shape[-1])  # (B, Z, Y, X, C)
bev_pool_v2后得到的特征是BZYXC, 那么下一句代码：bev_feat = torch.cat(bev_feat.unbind(dim=2), 1)
是把车辆前后Y维度合并到高度Z，变成了B,Z*Y,X,C，这句正确的做法应该是：
bev_feat.permute(0,4,1,2,3) # 变成 B,C,Z,Y,X
bev_feat = torch.cat(bev_feat.unbind(dim=2), 1) # 这样才是collapse Z, 维度变为B, C*Z, Y, X
bev_feat.permute(0, 2, 3, 1) # permute转回去，维度变为(B, Y, X, C*Z)

**Comment by northeastsquare** [comment]
没有问题，有一句话，我没看到。
https://github.com/HuangJunJie2017/BEVDet/blob/dev3.0/mmdet3d/ops/bev_pool_v2/bev_pool.py#L91
代码里已经把输出(B, Z, Y, X, C) permute变为(B, C, Z, Y, X)
所以最后，bev_feat = torch.cat(bev_feat.unbind(dim=2), 1)， 就是把高度Z维合并到C维了变成(B,C*Z, Y, X)，和LSS实现一致
